lib/internal/process/next_tick.js

```diff
@@ -210,43 +214,41 @@ function setupNextTick() {
     } while (tickInfo[kLength] !== 0);
   }
 
-  function TickObject(callback, args, domain) {
-    this.callback = callback;
-    this.domain = domain;
-    this.args = args;
-    this[async_id_symbol] = -1;
-    this[trigger_id_symbol] = -1;
-  }
-
-  function setupInit(tickObject, triggerId) {
-    tickObject[async_id_symbol] = ++async_uid_fields[kAsyncUidCntr];
-    tickObject[trigger_id_symbol] = triggerId || initTriggerId();
-    if (async_hook_fields[kInit] > 0) {
-      emitInit(tickObject[async_id_symbol],
-               'TickObject',
-               tickObject[trigger_id_symbol],
-               tickObject);
+  class TickObject {
+    constructor(callback, args, triggerId) {
+      var asyncId = ++async_uid_fields[kAsyncUidCntr];
+      this.callback = callback;
+      this.args = args;
+      this.domain = process.domain || null;
+      this[async_id_symbol] = asyncId;
+      this[trigger_id_symbol] = triggerId;
+      if (async_hook_fields[kInit] > 0)
```

Comment:
You can't move these other statements back into `nextTick()` without causing a performance regression from what I saw. I believe it's because of Crankshaft's inlining limits, but I did not want to introduce regressions at this moment (we will have to deal with it once TurboFan is everywhere).